### ISSUE TITLE
[directories] Ignore commented out backend config in 'Get Directories'

### ISSUE THREAD

**Issue by lnrdll** [abcxyz/guardian#219]
### TL;DR

Modify the [Get directories](https://github.com/abcxyz/guardian/blob/7427f64bb08f6a31d6700e676b1fbe0b3f4ca195/.github/actions/directories/action.yml#L42) step to exclude backend configurations that have been commented out.

### Detailed design

Currently, the [Get directories](https://github.com/abcxyz/guardian/blob/7427f64bb08f6a31d6700e676b1fbe0b3f4ca195/.github/actions/directories/action.yml#L42) step searches for the string `backend` in `.tf` files, but it doesn't consider whether the `backend` string is commented out or not.

The proposed update aims to modify the shell script to account for commented-out backend configurations.

[Current Implementation](https://github.com/abcxyz/guardian/blob/7427f64bb08f6a31d6700e676b1fbe0b3f4ca195/.github/actions/directories/action.yml#L50):

```shell
BACKEND_DIRS=$(find $TERRAFORM_DIRECTORY -name "*.tf" | xargs grep -l 'backend ".*" {' | xargs dirname | sed 's|[\.?][/]*||g' | sort | uniq)
```

Proposed Change:

```shell
BACKEND_DIRS=$(find $TERRAFORM_DIRECTORY -name "*.tf" | xargs awk '/backend ".*" {/ && ! /#[[:space:]]*backend ".*" {/ { print FILENAME }' | xargs dirname | sed 's|[\.?][/]*||g' | sort | uniq)
```
